slack-api-client/src/main/java/com/slack/api/rate_limits/metrics/impl/BaseMemoryMetricsDatastore.java

```diff
@@ -94,14 +123,27 @@ public ExecutorServiceProvider getExecutorServiceProvider() {
     @Override
     public void setExecutorServiceProvider(ExecutorServiceProvider executorServiceProvider) {
         this.executorServiceProvider = executorServiceProvider;
-        initializeCleanerExecutor();
+        if (this.cleanerEnabled) {
+            initializeCleanerExecutor();
```

Comment:
Skip the initialization (= starting a scheduled async code execution) if cleanerEnabled is false